crd/nodenetworkconfig/api/v1alpha/nodenetworkconfig.go

```diff
@@ -46,6 +47,8 @@ type NodeNetworkConfigSpec struct {
 	// +kubebuilder:validation:Optional
 	RequestedIPCount int64    `json:"requestedIPCount"`
 	IPsNotInUse      []string `json:"ipsNotInUse,omitempty"`
+	// AvailabilityZone contains the Azure availability zone for the virtual machine where network containers are placed.
+	AvailabilityZone string `json:"availabilityZone,omitempty"`
```

Comment:
let's add `// +kubebuilder:validation:Optional` since we expect that some NNCs might get created/updated without this field for some time

Reply:
added